File: client-app/src/app/layout/models/post.ts
import { ChatComment } from "./comment";
import { Like } from "./like";
export interface Post {
  id: string;
  username: string;
  displayName: string;
  caption: string;
  image: string;
  createdAt: Date;
  hostImage: string;
  comments: ChatComment[];
  likes: Like[];
  isHost?: boolean;
}
export class Post implements Post {
  constructor(init?: PostFormValues) {
    Object.assign(this, init);
  }
}
export class PostFormValues {
  id?: string = undefined;
  caption: string = "";
  image: any = null;

  constructor(post?: PostFormValues) {
    if (post) {
      this.id = post.id;
      this.caption = post.caption;
      this.image = post.image;
    }
  }
}

File: Domain/Post.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public AppUser Author { get; set; }
        public string Caption { get; set; }
        public string Image { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Like> Likes { get; set; } = new List<Like>();
    }
}